File: datasets/php-keep-code_davinci_001_temp_0.2/HumanEval_149_sorted_list_sum.php
<?php
// Write a function that accepts a list of strings as a parameter,
// deletes the strings that have odd lengths from it,
// and returns the resulted list with a sorted order,
// The list is always a list of strings and never an array of numbers,
// and it may contain duplicates.
// The order of the list should be ascending by length of each word, and you
// should return the list sorted by that rule.
// If two words have the same length, sort the list alphabetically.
// The function should return a list of strings in sorted order.
// You may assume that all words will have the same length.
// For example:
// assert list_sort(["aa", "a", "aaa"]) => ["aa"]
// assert list_sort(["ab", "a", "aaa", "cd"]) => ["ab", "cd"]
function sorted_list_sum($lst) {
    $lst = array_filter($lst, function($x) {
        return strlen($x) % 2 == 0;
    });
    usort($lst, function($a, $b) {
        return strlen($a) == strlen($b) ? strcmp($a, $b) : strlen($a) - strlen($b);
    });
    return $lst;
}


function candidate(...$args) {
    return sorted_list_sum(...$args);
}

function test(): void {
    if (candidate(array("aa", "a", "aaa")) !== array("aa")) { throw new Exception("Test failed!"); }
    if (candidate(array("school", "AI", "asdf", "b")) !== array("AI", "asdf", "school")) { throw new Exception("Test failed!"); }
    if (candidate(array("d", "b", "c", "a")) !== array()) { throw new Exception("Test failed!"); }
    if (candidate(array("d", "dcba", "abcd", "a")) !== array("abcd", "dcba")) { throw new Exception("Test failed!"); }
    if (candidate(array("AI", "ai", "au")) !== array("AI", "ai", "au")) { throw new Exception("Test failed!"); }
    if (candidate(array("a", "b", "b", "c", "c", "a")) !== array()) { throw new Exception("Test failed!"); }
    if (candidate(array("aaaa", "bbbb", "dd", "cc")) !== array("cc", "dd", "aaaa", "bbbb")) { throw new Exception("Test failed!"); }
}

test();

File: datasets/originals-with-cleaned-vocab-lua/HumanEval_149_sorted_list_sum.py
from typing import List

def sorted_list_sum(lst: List[str]) -> List[str]:
    """Write a function that accepts a table of strings as a parameter,
    deletes the strings that have odd lengths from it,
    and returns the resulted table with a sorted order,
    The table is always a table of strings and never a table of numbers,
    and it may contain duplicates.
    The order of the table should be ascending by length of each word, and you
    should return the table sorted by that rule.
    If two words have the same length, sort the table alphabetically.
    The function should return a table of strings in sorted order.
    You may assume that all words will have the same length.
    For example:
    >>> list_sort(["aa", "a", "aaa"]) 
    ["aa"]
    >>> list_sort(["ab", "a", "aaa", "cd"]) 
    ["ab", "cd"]
    """
    ### Canonical solution below ###
    lst.sort()
    new_lst = []
    for i in lst:
        if len(i)%2 == 0:
            new_lst.append(i)
    return sorted(new_lst, key=len)

### Unit tests below ###
def check(candidate):

    # Check some simple cases
    assert candidate(["aa", "a", "aaa"]) == ["aa"]
    assert candidate(["school", "AI", "asdf", "b"]) == ["AI", "asdf", "school"]
    assert candidate(["d", "b", "c", "a"]) == []
    assert candidate(["d", "dcba", "abcd", "a"]) == ["abcd", "dcba"]

    # Check some edge cases that are easy to work out by hand.
    assert candidate(["AI", "ai", "au"]) == ["AI", "ai", "au"]
    assert candidate(["a", "b", "b", "c", "c", "a"]) == []
    assert candidate(['aaaa', 'bbbb', 'dd', 'cc']) == ["cc", "dd", "aaaa", "bbbb"]

def test_check():
    check(sorted_list_sum)
